### ISSUE TITLE
Multi-GPU can not speed up training

### ISSUE THREAD

**Issue by wuqi930907** [dbolya/yolact#257]
Thanks for your work.
I want to train my own model with multi-gpus, but something seems abnormal.
I have trained one model with single gpu, it costed 3 days(Iteration times:160000, batch_size:5).
Then I modified  Iteration times and batch_size to 40000 and 32,it also costed 3 days to let the training complete with 4 gpus.
Could you give me some suggestions, how to speed up traing with multi-gpu?
Thank you!

**Comment by wuqi930907** [comment]
By the way,following is my command:
single gpu:CUDA_VISIBLE_DEVICES=3 python3 train.py --config=yolact_plus_base_config --batch_size=5
4 gpus:CUDA_VISIBLE_DEVICES=0,1,2,3 python3 train.py --config=yolact_plus_base_config --batch_size=32
and I have modified the iteration times in config.py(single gpu:160000, 4 gpus:40000)

**Comment by dbolya** [comment]
It should scale the number of iterations automatically, so no need to manually modify the number of iterations. The lack if speedup is weird though, since everything else is right and if you also also modified the number of iterations it should actually have been tried to do far less.

Do you see a message at the start of training saying "Batch size of x blah blah scaling parameters by 4" when you use multiple GPUs? And what's the difference in "timer" between 1 and 4 gpus?

I get a roughly 3-3.5x speedup when I use 4 gpus as opposed to 1.

**Comment by wuqi930907** [comment]
Thank you for your reply, following is my logs:
single gpu:
Scaling parameters by 0.62 to account for a batch size of 5.
Per-GPU batch size is less than the recommended limit for batch norm. Disabling batch norm.
loading annotations into memory...
Done (t=0.31s)
creating index...
index created!
loading annotations into memory...
Done (t=0.01s)
creating index..

4 gpus:
Multiple GPUs detected! Turning off JIT.
Scaling parameters by 4.00 to account for a batch size of 32.
loading annotations into memory...
Done (t=0.06s)
creating index...
index created!
loading annotations into memory...
Done (t=0.01s)
creating index...

And my environment:
Ubuntu16.04
CUDA 10.1
cudnn 7.6.4
Pytorch 1.3
python3.5
TitanXp

Thanks

**Comment by dbolya** [comment]
Have you got it to speed up training yet? I'm not sure I can recommend a fix since it's likely something specific to your machine.

**Comment by voidrank** [comment]
Hi @wuqi930907 @dbolya ,

I may find out the solution to this issue: I replaced the CustomeDataParallel to original DataParallel and training speed turned normal. I'm using Pytorch1.5, so I guess this issue may be caused by different implementation of DataParallel or something related.
